### PR TITLE
allow for array alphabet

### PR DESCRIPTION
**Pull request by aakilfernandes**
Hi, this is a PR which allows for alphabets as arrays. It may contain a bit of a performance hit since apparently treating strings like arrays cause a bit of a performance hit.

I needed this to work with a base system with multi-character encoding systems (ie treating bip39 like an encoding scheme)

### CODE REVIEW

**Comment by jprichardson** [comment]
Interesting. What's the use case? Looks bip39-esque..

**Comment by aakilfernandes** [comment]
Yup uses the bip39 words. But bip39 has some checksum stuff rolled into it so this is just for a generic converter.

Use case is for a 2fa wallet where users need to put some code hash into a smart contract to unlock fund.

**Comment by dcousens** [comment]
ACK-ish,  but not with the inline BIP39 alphabet included.. lets keep the test cases minimal.

File: index.js [hunk 1]
```diff
@@ -9,11 +9,11 @@
 module.exports = function base (ALPHABET) {
   var ALPHABET_MAP = {}
   var BASE = ALPHABET.length
-  var LEADER = ALPHABET.charAt(0)
+  var LEADER = ALPHABET[0]
```
Comment (dcousens) [inline index.js]: I don't believe this is guaranteed to work as well as we think?
Can we get some reference to `.charAt` up here first,  and if necessary,  split `ALPHABET` prior?

Comment (dcousens) [inline index.js]: Looks good via https://stackoverflow.com/questions/5943726/string-charatx-or-stringx#5943760

**Comment by dcousens** [comment]
ACK,  but I'm going to reduce/remove the included massive alphabet,  and also going to change the API to instead default a `delimiter` parameter

**Pull request merged**